Dockerfile.apache

```diff
@@ -8,12 +8,20 @@
 # Contributors:
 #   Red Hat, Inc. - initial API and implementation
 
-FROM docker.io/httpd:2.4.43-alpine
+FROM docker.io/node:8.16.2-alpine as builder
+
+COPY package.json /dashboard/
+COPY yarn.lock /dashboard/
+WORKDIR /dashboard
+RUN yarn install --ignore-optional
+COPY . /dashboard/
+RUN yarn build && yarn test
 
+FROM docker.io/httpd:2.4.43-alpine
 RUN sed -i 's|    AllowOverride None|    AllowOverride All|' /usr/local/apache2/conf/httpd.conf && \
     sed -i 's|Listen 80|Listen 8080|' /usr/local/apache2/conf/httpd.conf && \
     mkdir -p /var/www && ln -s /usr/local/apache2/htdocs /var/www/html && \
     chmod -R g+rwX /usr/local/apache2 && \
     echo "ServerName localhost" >> /usr/local/apache2/conf/httpd.conf
-COPY target/dist /usr/local/apache2/htdocs/dashboard
-RUN sed 's/<![CDATA[<base href="\/">]]/<![CDATA[<base href="\/dashboard">]]/g'  /usr/local/apache2/htdocs/dashboard/index.html
+COPY --from=builder /dashboard/target/dist/ /usr/local/apache2/htdocs/dashboard
+RUN sed -i 's/<base href="\/">/<base href="\/dashboard\/"/g'  /usr/local/apache2/htdocs/dashboard/index.html
```

Comment:
You can make this slightly easier to read if you use the `-r` flag for extended regex. 

Then using a different field delimiter you can do:

```
RUN sed -i -r -e 's#<base href="/">#<base href="/dashboard/"#g'  /usr/local/apache2/htdocs/dashboard/index.html 
```